index.js

```diff
@@ -2,33 +2,31 @@
 
 var jingleStanza = require('jxt').createRegistry();
 
-
 jingleStanza.use(require('jxt-xmpp-types'));
-jingleStanza.use(require('./stanzas/iq'));
-jingleStanza.use(require('./stanzas/message'));
-jingleStanza.use(require('./stanzas/error'));
-jingleStanza.use(require('./stanzas/jingle'));
-jingleStanza.use(require('./stanzas/rtp'));
-jingleStanza.use(require('./stanzas/file'));
-jingleStanza.use(require('./stanzas/iceUdp'));
-jingleStanza.use(require('./stanzas/extdisco'));
-jingleStanza.use(require('./stanzas/jingleMessage'));
+jingleStanza.use(require('jxt-xmpp'));
 
 module.exports = {
-    getXml: function (data) {
-        var Iq = jingleStanza.getIq();
-        return new Iq(data);
-    },
+  getXml: function (data) {
+    var Iq = jingleStanza.getIq();
+    return new Iq(data);
+  },
+
+  getMessageXml: function (data) {
+    var Message = jingleStanza.getMessage();
+    return new Message(data);
+  },
 
-    getMessageXml: function (data) {
-        var Message = jingleStanza.getMessage();
-        return new Message(data);
-    },
+  getData: function (stanza) {
+    if (typeof stanza === 'string') {
+      return jingleStanza.parse(stanza);
+    }
 
-    getData: function (stanza) {
-        if (typeof stanza !== 'string') {
-            stanza = stanza.toString();
-        }
-        return jingleStanza.parse(stanza);
+    // Indicates the stanza is an Element rather than a DOMElement,
+    // which is what JXT expects so let jxt parse it instead
+    if (!stanza.localName && !stanza.namespaceURI) {
+      return jingleStanza.parse(stanza.toString());
```

Comment:
This is the bit that's getting back the optimization - instead of doing this for all stanzas, only do it for stanzas that aren't DOMElement types.